src/bijectors/simplex.jl

```diff
@@ -213,12 +200,10 @@ function simplex_logabsdetjac_gradient(x::AbstractMatrix)
     return g
 end
 
-function simplex_link_jacobian(
-    x::AbstractVector{T}, ::Val{proj}=Val(true)
-) where {T<:Real,proj}
+function simplex_link_jacobian(x::AbstractVector{T}) where {T<:Real}
     K = length(x)
     @assert K > 1 "x needs to be of length greater than 1"
-    dydxt = similar(x, length(x), length(x))
+    dydxt = similar(x, K, K - 1)
```

Comment:
Maybe merge it with the next line while you're at it to make it clearer that `dydx` is initialized with zeros (I had missed that initially when reading the code)?
```suggestion
    dydxt = fill!(similar(x, K, K - 1), 0)
```

Reply:
(BTW the `similar` + `AbstractVector` argument wrongly suggests that the code will work for arbitrary inputs - but the loops below assume that everything uses 1-based indices)

Reply:
Agreed, but this should be fixed in a separate PR.